src/Sonarr.Http/PagingResource.cs

```diff
@@ -38,7 +38,11 @@ public PagingResource(PagingRequestResource requestResource)
 
     public static class PagingResourceMapper
     {
-        public static PagingSpec<TModel> MapToPagingSpec<TResource, TModel>(this PagingResource<TResource> pagingResource, string defaultSortKey = "Id", SortDirection defaultSortDirection = SortDirection.Ascending)
+        public static PagingSpec<TModel> MapToPagingSpec<TResource, TModel>(
+            this PagingResource<TResource> pagingResource,
+            string defaultSortKey = "Id",
+            SortDirection defaultSortDirection = SortDirection.Ascending,
+            HashSet<string> allowedSortKeys = null)
```

Comment:
To avoid future issues should this be a required parameter? Otherwise passing in nothing/null/empty hash set will use whatever key is passed in (which could be anything from an API client).